src/main/java/uk/gov/cslearning/record/service/DefaultInviteService.java

```diff
@@ -18,28 +20,39 @@
     private final InviteDtoFactory inviteDtoFactory;
     private final InviteRepository inviteRepository;
     private final EventService eventService;
+    private final IdentityService identityService;
 
-    public DefaultInviteService(InviteFactory inviteFactory, InviteDtoFactory inviteDtoFactory, InviteRepository inviteRepository, EventService eventService){
+    public DefaultInviteService(InviteFactory inviteFactory, InviteDtoFactory inviteDtoFactory, InviteRepository inviteRepository, EventService eventService, IdentityService identityService){
         this.inviteFactory = inviteFactory;
         this.inviteDtoFactory = inviteDtoFactory;
         this.inviteRepository = inviteRepository;
         this.eventService = eventService;
+        this.identityService = identityService;
     }
 
     @Override
     public Collection<InviteDto> findByEventId(String eventId){
-        Collection<InviteDto> result = inviteRepository.findByEventId(eventId)
+        Collection<InviteDto> result = inviteRepository.findAllByEventUid(eventId)
                 .stream().map(
                         inviteDtoFactory::create
                 ).collect(Collectors.toList());
 
-
         return result;
     }
 
     @Override
-    public InviteDto save(InviteDto inviteDto){
+    public Optional<InviteDto> findInvite(int id){
+        return inviteRepository.findById(id).map(inviteDtoFactory::create);
+    }
+
+    @Override
+    public Optional<InviteDto> save(InviteDto inviteDto){
         Event event = eventService.getEvent(Paths.get(inviteDto.getEvent().getPath()).getFileName().toString(), inviteDto.getEvent().getPath());
-        return inviteDtoFactory.create(inviteRepository.save(inviteFactory.create(inviteDto, event)));
+
+        Optional<Invite> invite = inviteRepository.findByEventIdAndLearnerEmail(event.getId(), inviteDto.getLearnerEmail());
```

Comment:
Why do you a check to see if the invite exists here? I don't think it's necessary. If it exists the operation will be a no op and the look up is potentially expensive.

```
    @Override
    public Optional<InviteDto> save(InviteDto inviteDto){
        Event event = eventService.getEvent(Paths.get(inviteDto.getEvent().getPath()).getFileName().toString(), inviteDto.getEvent().getPath());

        return Optional.of(inviteDtoFactory.create(inviteRepository.save(inviteFactory.create(inviteDto, event))));
    }

```